command/jsonstate/state.go

```diff
@@ -84,6 +84,13 @@ type resource struct {
 	// unknown values are omitted or set to null, making them indistinguishable
 	// from absent values.
 	AttributeValues attributeValues `json:"values,omitempty"`
+
+	// DependsOn contains a list of the resource's dependencies. The entries are
+	// addresses relative to the resouce.
```

Comment:
I think, based on how this is implemented, that these are relative to the containing module, giving addresses like `aws_instance.bar`.

Reply:
Yeah, this was something that I was wondering too. Do we _ever_ track these in state across modules? Seeing as inter-module dependencies do exist between resources, will there ever be resource paths that include modules?

Reply:
Dependencies between modules are always indirect: rather than `aws_instance.foo` depending on `module.baz.aws_instance.other` directly, there's an indirection through an output of `module.baz`, which would appear in this list as something like `module.baz.output_name`. There's a manifestation of that output living within the module too, which can in turn depend on a resource in that child module, but that only ever appears in the in-memory dependency graph, and is never persisted in state snapshots or saved plan files.

(The only reason we retain the `depends_on` for _resources_ in the state is so that we can still honor them when the resource has been removed from configuration and Terraform needs to destroy it. Non-root output values never outlive a single run, so we can always get those dependencies from the configuration.)

Reply:
Ah, yeah, it looks like my brain disengaged whilst typing. Thank you, good catch!